hplsql/src/main/java/org/apache/hive/hplsql/udf/Udf.java

```diff
@@ -37,77 +33,59 @@
 import org.apache.hive.hplsql.Scope;
 import org.apache.hive.hplsql.Var;
 import org.apache.hive.hplsql.executor.QueryExecutor;
-import org.apache.thrift.TException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-@Description(name = "hplsql", value = "_FUNC_('query' [, :1, :2, ...n]) - Execute HPL/SQL query", extended = "Example:\n" + " > SELECT _FUNC_('CURRENT_DATE') FROM src LIMIT 1;\n")
+@Description(name = "hplsql", value = "_FUNC_('query' [, :1, :2, ...n], 'storedProcedure') - Execute HPL/SQL query",
+        extended = "Example:\n" + " > SELECT _FUNC_('CURRENT_DATE') FROM src LIMIT 1;\n")
 @UDFType(deterministic = false)
 public class Udf extends GenericUDF {
   public static String NAME = "hplsql";
-  transient Exec exec;
-  StringObjectInspector queryOI;
-  ObjectInspector[] argumentsOI;
-  private String functionDefinition;
+  private final transient Exec exec = new Exec();
+  private StringObjectInspector queryOI;
+  private ObjectInspector[] argumentsOI;
+  private StringObjectInspector funcDefOI;
+  private String functionDefinition = null;
+  private static final Logger LOG = LoggerFactory.getLogger(Udf.class.getName());
```

Comment:
nit.: static field declarations especially logger is usually the first in the class.